src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/PersonsTest.kt

```diff
@@ -142,6 +143,65 @@ class PersonsTest: AbstractTest() {
         }
     }
 
+    /**
+     * Tests overriding vacation days for users via v/1/persons/{personId} -endpoint PUT method. Only admin-roles should be allowed to perform this action.
+     */
+    @Test
+    fun overrideVacationDaysTest() {
+        createTestBuilder().use { testBuilder ->
+            val person = testBuilder.userA.persons.getPersons(active = false).find { it.id == 2 }!!
+            val newPerson = Person(
+                id = person.id,
+                firstName = person.firstName,
+                lastName = person.lastName,
+                email = person.email,
+                monday = person.monday,
+                tuesday = person.tuesday,
+                wednesday = person.wednesday,
+                thursday = person.thursday,
+                friday = person.friday,
+                saturday = person.saturday,
+                sunday = person.sunday,
+                active = person.active,
+                unspentVacations = 32,
+                spentVacations = 37,
+                minimumBillableRate = person.minimumBillableRate,
+                language = person.language,
+                startDate = person.startDate
+            )
+            val updatedPerson = testBuilder.manager.persons.updatePerson(
+                personId = newPerson.id,
+                person = newPerson
+            )
+
+            assertEquals(32, updatedPerson.unspentVacations)
+            assertEquals(37, updatedPerson.spentVacations)
+
+            testBuilder.userA.persons.assertUpdateFail(
+                person = newPerson,
+                expectedStatus = 401
+            )
+
+            try {
+                testBuilder.admin.persons.updatePerson(
+                    personId = newPerson.id,
+                    person = newPerson
+                )
+            } catch (e: ClientException) {
+                assertNotEquals(401, e.statusCode, "Update failed with 401 status")
+            }
```

Comment:
Existing `TestBuilderResource`s have reference methods for asserting client/server failures in a more cleaner way.

For example this would be

```kotlin
fun assertUpdateFail(expectedStatus: Int, personId: Int, person: Person) {
    try {
        api.updatePerson(
            personId = personId,
            person = person
        )
    } catch (e: ClientException) {
        assertClientExceptionStatus(expectedStatus, e)
    }
}
```